docs/guide.rst

```diff
@@ -873,6 +873,34 @@ copied to the guest.
     ``finish`` steps yet.
 
 
+.. _when-config:
+
+Conditional step configuration
+------------------------------
+
+.. versionadded:: 1.39
+
+Sometimes, the plan is expected to cover a broad set of environments;
+however, some step configurations may not be applicable everywhere.
+While :ref:`/spec/core/adjust` can be used to construct the plan
+in this way, it soon becomes difficult to read.
+
+Using the ``when`` key makes it easier to restrict a step configuration
+to run only if any of the specified rules matches.
```

Comment:
so this means I can also use something like

```
when:
  - distro == fedora && trigger = rhel-ci
  - distro == centos && trigger = fedora-ci
```
?

Reply:
I would appreaciate more examples here, to get the feel of what can be done here

Reply:
https://github.com/teemtee/tmt/commit/0fedbe1e4266707fcf2d19adf9926ffb58ac097e